### PR TITLE
Update swig

### PR DESCRIPTION
**Pull request by aprokop**
First part of @sethrj work. This only updates the wrappers with the most recent SWIG and fixes the build for that.

### CODE REVIEW

File: src/tpetra/test/test_tpetra_import_export.f90 [hunk 1]
```diff
@@ -170,7 +171,7 @@ program test_TpetraImportExport
       !                     [n-1  2n-1  3n-1  4n-1  5n-1]
       call mv_with_neighbors%doImport(mv_mine, importer, TpetraREPLACE)
       do j = 1, num_vecs
-        a = mv_with_neighbors%getData(j)
+        a => mv_with_neighbors%getData(j)
```
Comment (sethrj) [inline src/tpetra/test/test_tpetra_import_export.f90]: Probably want to change the function name to something like `getView`

Comment (aprokop) [reply src/tpetra/test/test_tpetra_import_export.f90]: The name is coming from Tpetra, don't necessarily want to change that.

Comment (sethrj) [reply src/tpetra/test/test_tpetra_import_export.f90]: If we want to distinguish functions that return array pointers from functions that return allocated arrays, won't we have to change some function names?

Comment (tjfulle) [reply src/tpetra/test/test_tpetra_import_export.f90]: FWIW, I might lean toward changing the name to `getView` to have consistency in ForTrilinos.  Or, keep `getData` and write a `getView` function that merely wraps `getData`.

**Pull request merged**